generate.c

```diff
@@ -1256,18 +1256,32 @@ smartypants(int c, int *flags, MMIOT *f)
 static int
 mathhandler(MMIOT *f, int e1, int e2)
 {
-    int i = 0;
-
-    while(peek(f, ++i) != EOF) {
-        if (peek(f, i) == e1 && peek(f, i+1) == e2) {
-	    cputc(peek(f,-1), f);
-	    cputc(peek(f, 0), f);
-	    while ( i-- > -1 )
-		cputc(pull(f), f);
-            return 1;
-        }
-    }
-    return 0;
+	int i = 0;
+	while(peek(f, ++i) != EOF) {
+		if (peek(f, i) == e1 && peek(f, i+1) == e2) {
+			int charcount = (i+1) + 2;
+
+			char* math = (char*)malloc(sizeof(char)*(charcount+1));
+			math[0] = peek(f, -1);
+			math[1] = peek(f, 0);
+
+			while ( i-- > -1 )
+				math[charcount-i-2] = pull(f);
+			math[charcount] = '\0';
+
+			char* edit;
+			if ( f->cb && f->cb->e_latex && (edit = (*f->cb->e_latex)(math, charcount+1, NULL)) ) {
```

Comment:
I am not sure about size. Should `size` passed into the callback with counted `\0` symbol, like here?

Reply:
Also, is `\0` necessary at all?